pact-request-api/Pact/Core/Command/Util.hs

```diff
@@ -0,0 +1,283 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- |
+-- Module      :  Pact.Types.Util
+-- Copyright   :  (C) 2016 Stuart Popejoy
+-- License     :  BSD-style (see the file LICENSE)
+-- Maintainer  :  Stuart Popejoy <stuart@kadena.io>
+--
+-- Utility types and functions.
+--
+module Pact.Core.Command.Util
+  ( -- | Text parsing
+    ParseText(..)
+  , fromText, fromText'
+  -- | JSON helpers
+  , satisfiesRoundtripJSON, roundtripJSONToEither
+  , lensyToJSON, lensyParseJSON, lensyOptions, lensyConstructorToNiceJson
+  , unsafeFromJSON, outputJSON
+  , fromJSON'
+  , enableToJSON
+  -- | Base 16 helpers
+  , parseB16JSON, parseB16Text, parseB16TextOnly
+  , toB16Text
+  , B16JsonBytes(..)
+  -- | Base64Url helpers
+  , encodeBase64UrlUnpadded, decodeBase64UrlUnpadded
+  , parseB64UrlUnpaddedText, parseB64UrlUnpaddedText'
+  , toB64UrlUnpaddedText, fromB64UrlUnpaddedText
+  , B64JsonBytes(..)
+  -- | AsString
+  , AsString(..), asString'
+  -- | MVar-as-state utils
+  , modifyMVar', modifyingMVar, useMVar
+  -- | Miscellany
+  , tShow, maybeDelim
+  , maybeToEither
+  -- | Wrapping utilities
+  , rewrap, rewrapping, wrap, unwrap
+  -- | Arbitrary generator utils
+  -- , genBareText
+  ) where
+
+import Data.Aeson
+import Data.Aeson.Types
+import GHC.Generics
+import Data.ByteString (ByteString)
+-- import qualified Data.Attoparsec.Text as AP
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Base16 as B16
+import qualified Data.ByteString.Base64.URL as B64URL
+import qualified Data.ByteString.Lazy.Char8 as BSL8
+import qualified Data.ByteString.Short as SB
+-- import qualified Text.Trifecta as Trifecta
+import Data.Char
+-- import Data.Either (isRight)
+import Data.Hashable (Hashable)
+import Data.Word
+import Data.Text (Text,pack,unpack)
+import Data.Text.Encoding
+-- import Control.Applicative ((<|>))
+import Control.Concurrent
+import Control.Lens hiding (Empty, elements, (.=))
+
+-- import Pact.Types.Parser (style, symbols)
+
+import GHC.Stack (HasCallStack)
```

Comment:
Uh, is this what we want?

Reply:
There are many things in this module we don't need. Will clean those up.